mitmproxy/net/tcp.py

```diff
@@ -915,19 +821,18 @@ class TCPServer:
     request_queue_size = 20
 
     def __init__(self, address):
-        self.address = Address.wrap(address)
+        self.address = address
         self.__is_shut_down = threading.Event()
         self.__shutdown_request = False
-        self.socket = socket.socket(self.address.family, socket.SOCK_STREAM)
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
As you mentioned, this is why IPv6 binding fails. We probably should implement the strategy proposed in https://github.com/mitmproxy/mitmproxy/pull/1995 here: Try IPv6 with V6ONLY=false, and fall back to IPv4 if that fails. We can do that in a separate PR if you want.